test/state-component-mocha.js

```diff
@@ -0,0 +1,139 @@
+// state-component-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var noflo = require('noflo');
+var stateComponent = require('../components/state-component.js');
+
+describe('promise-component', function() {
+    it("should reject undefined definition", function() {
+        return Promise.resolve().then(stateComponent).should.be.rejected;
+    });
+    it("should reject empty definition", function() {
+        return Promise.resolve({}).then(stateComponent).should.be.rejected;
+    });
+    it("should reject empty inPorts", function() {
+        return Promise.resolve({inPorts:{}}).then(stateComponent).should.be.rejected;
+    });
+    it("should accept array of inPort names", function() {
```

Comment:
That is not what this test is covering. Other tests both in this file and in the promise-component-mocha test that.